### PR TITLE
homepage hero bug fixes

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?
- Closes https://github.com/mitodl/hq/issues/4463

### Description (What does it do?)
This PR addresses several issues in the homepage hero section:
- Fixes hover shift on the search icon. *This bug affected search page, too, and all inputs with icon adornments.*
- Fixes hover effect—darken the border, not thicken.
- Makes hero image match figma on full screen.
     - Figma does not indicate the behavior for screens between mobile and desktop. The image and searchbox scale.
- Shifts hero image so it slightly overflows the content container, as in figma. 
- Adjusts the font size and padding of the mobile view search input.
     - `size="hero"` for input should now match the new figma "mobile (hero)" input.

### Screenshots (if appropriate):

*Note: The text colors in this movie are slightly wrong. They are corrected in the screenshots. The movie primarily shoes sizing behavior.*
https://github.com/mitodl/mit-open/assets/9010790/6714b2bf-087d-4324-8df3-5a275611ade3

Screenshot 2024-06-07 at 5 31 13 PM

Screenshot 2024-06-07 at 5 31 38 PM

Screenshot 2024-06-07 at 5 31 43 PM

Screenshot 2024-06-07 at 4 46 50 PM

### How can this be tested?
Load the homepage and try it at different sizes.

### CODE REVIEW

**Comment by mbilalmughal** [comment]
@ChristopherChudzicki The Hero section UI looks good to me, and the video displays the correct portions. Just one small thing: please remove the underline from the search pills.


Another nice-to-have improvement would be to slightly reduce the heading font size on medium devices, similar to what we did for mobile, but not as small. Also, reduce the space between the hero section and the search bar by half on medium devices media queries. This will help adjust the layout nicely. rest is all good 👍 

image

**Review by jkachel — CHANGES_REQUESTED** [review]
Looks pretty good to me other than the things mentioned in [Bilal's comments](https://github.com/mitodl/mit-open/pull/1034#issuecomment-2158417419) (as he beat me to the punch).

File: frontends/mit-open/src/GlobalStyles.tsx [hunk 1]
```diff
@@ -29,11 +29,6 @@ const pageCss = css`
 
   a {
     text-decoration: none;
-
-    &:hover {
-      text-decoration: underline;
-    }
```
Comment (ChristopherChudzicki) [inline frontends/mit-open/src/GlobalStyles.tsx:35]: Removing this everywhere. Where we want it, it's added by our `Link` styles.

File: frontends/ol-components/src/components/ThemeProvider/chips.tsx [hunk 2]
```diff
@@ -76,6 +76,9 @@ const chipComponent: NonNullable<ThemeOptions["components"]>["MuiChip"] = {
       style: {
         borderColor: colors.silverGrayLight,
         color: colors.darkGray1,
+        "&.Mui-focusVisible": {
+          backgroundColor: "transparent",
+        },
```
Comment (ChristopherChudzicki) [inline frontends/ol-components/src/components/ThemeProvider/chips.tsx:81]: This and other changes here avoid mui's default focus-visible style, which was problematic for one of our variants.

**Review by jkachel — APPROVED** [review]
LGTM 👍

**Pull request merged**